django/core/validators.py

```diff
@@ -397,8 +397,12 @@ class StepValueValidator(BaseValidator):
     message = _("Ensure this value is a multiple of step size %(limit_value)s.")
     code = "step_size"
 
+    def __init__(self, limit_value, offset=None, message=None):
+        super().__init__(limit_value, message)
+        self.offset = 0 if offset is None else offset
+
     def compare(self, a, b):
-        return not math.isclose(math.remainder(a, b), 0, abs_tol=1e-9)
+        return not math.isclose(math.remainder(a + self.offset, b), 0, abs_tol=1e-9)
```

Comment:
I think there is a slight problem with this implementation when `cleaned == offset`. The test `(StepValueValidator(Decimal(2.0), offset=Decimal(0.1)), Decimal(0.1), None),` should pass, this should be the fix:

```suggestion
        return not math.isclose(math.remainder(a - self.offset, b), 0, abs_tol=1e-9)
```

Reply:
well spotted!
Thanks @jmsmkn